viewing_party/party.py

```diff
@@ -0,0 +1,187 @@
+                if watched_title == friend_watched_title:
+                    watched_title_found = True
+        
+        #I did not find it in the friends, so append it
+        if watched_title_found == False:
+            unique_dict.append(record)
+    
+    return unique_dict
+    
+def get_friends_unique_watched(user_data):
+    #Set up a variable to hold the unique titles
+    unique_dict = []
+    friends_dict = []
+
+    #Loop through all the friends watched data and create a single list of dictionaries
+    #with those records 
+    for friend in user_data["friends"]:
+        for friend_watched_record in friend["watched"]:
+            if friend_watched_record not in friends_dict:
+                friends_dict.append(friend_watched_record)
+
+
+    #Loop through the friends_dict  to compare to user_data_watched
+    for friend_watched_record in friends_dict:
+        if friend_watched_record not in user_data["watched"]:
+            unique_dict.append(friend_watched_record)
+    
+    return unique_dict 
+
+def get_available_recs(user_data):
+    #Set up a variable to hold friends suscriptions
+    friends_suscrip_dict = []
+    recommendation_dict = []
+    
+    #Loop through friends watched data and create a single list of dictionaries with those records. (title and host as keys with their values)
+    for friend in user_data["friends"]:
+        for friend_watched_record in friend["watched"]:
+            if friend_watched_record not in friends_suscrip_dict:
+                friends_suscrip_dict.append(friend_watched_record)
+    
+    #Loop through the friends dictionary to recomend user_data suscriptions. 
+    for friend_watched_record in friends_suscrip_dict:
+        #does the user have the service from the friend recommendation?
+        if friend_watched_record["host"] in user_data["subscriptions"]:
+            #has the user not seen the movie?
+            already_seen = False
+            for record in user_data["watched"]:
+                #already_seen = False
+                if record["title"] == friend_watched_record["title"]:
+                    already_seen = True
+            if not already_seen:
+                recommendation_dict.append(friend_watched_record)
+                #user_data["watched"].append(friend_watched_record)
+
+    return recommendation_dict  
+
+def get_new_rec_by_genre(user_data):
+
+    #Set up variables to hold friends or reference to friends_dict for comaprison
+    friends_dict = []
+    recommendation_list = []
+
+    #If the user hasnt watch any movies return an empty list.
+    if len(user_data["watched"]) == 0:
+        return []
+
+    #Loop through friends watched data and create a single list of dicitionaries with those records 
+    for friend in user_data["friends"]:
+        for friend_watched_record in friend["watched"]:
+            if friend_watched_record not in friends_dict:
+                friends_dict.append(friend_watched_record)    
```

Comment:
This section (151-154) is the same as 119-122 above.  I recommend pulling sections like this out into helper functions.